### PR TITLE
ast-grep: init at 0.5.2

### PR DESCRIPTION
**Pull request by astratagem**
###### Description of changes

[ast-grep/ast-grep: A fast and polyglot tool for code searching, linting, rewriting at large scale](https://github.com/ast-grep/ast-grep)

> ast-grep is a AST-based tool to search code by pattern code. Think it as your old-friend `grep` but it matches AST nodes instead of text. You can write patterns as if you are writing ordinary code. It will match all code that has the same syntactical structure. You can use `$` sign + upper case letters as wildcard, e.g. `$MATCH`, to match any single AST node. Think it as REGEX dot `.`, except it is not textual.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests]([https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests])
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
@ofborg eval

**Comment by montchr** [comment]
Result of `nixpkgs-review pr 231553` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ast-grep</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 231553` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ast-grep</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
Code looks good.
Binary runs.

File: pkgs/development/tools/misc/ast-grep/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
```
Comment (bjornfor) [inline pkgs/development/tools/misc/ast-grep/default.nix:4]: Isn't the nixpkgs style to have commas at the beginning?

Comment (Janik-Haag) [reply pkgs/development/tools/misc/ast-grep/default.nix:4]: Yes you are right I have overlooked that.

Comment (montchr) [reply pkgs/development/tools/misc/ast-grep/default.nix]: Reformatted with `nixpkgs-fmt`

**Review by Janik-Haag — APPROVED** [review]

**Comment by montchr** [comment]
Looks like there's a new version already. I'll update.

**Comment by montchr** [comment]
Result of `nixpkgs-review pr 231553` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ast-grep</li>
  </ul>
</details>

**Comment by bjornfor** [comment]
Commit message still says 0.5.1.

**Comment by Janik-Haag** [comment]
No it doesn't

**Comment by bjornfor** [comment]
> No it doesn't

Right. Sorry.

**Pull request merged**